src/OpenTelemetry.Extensions.Enrichment.AspNetCore/BaseAspNetCoreEnricher.cs

```diff
@@ -0,0 +1,61 @@
+// <copyright file="BaseAspNetCoreEnricher.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using Microsoft.AspNetCore.Http;
+
+namespace OpenTelemetry.Extensions.Enrichment.AspNetCore;
+
+/// <summary>
+/// Represents base ASP.NET Core enricher class.
+/// </summary>
+/// <typeparam name="T">Typo of enrichment bag to store enrichment tags.</typeparam>
+public abstract class BaseAspNetCoreEnricher<T>
```

Comment:
I'm guessing this was introduced so that it can be reused later with metrics and/or logs? My suggestion would be to remove it and then put it back if it ends up being needed/useful. I don't _think_ it would be breaking to do that. Why? Not 100% sure on this, but two levels of indirection. When a call happens it will have to be resolved (callvirt) using the heirarchy. Derived class, AspNetCoreTraceEnricher, or BaseAspNetCoreEnricher<T> may have the final implementation. So dropping one level might be faster. But also it just doesn't seem to be used so it is extra public API 😄